mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/search/definition/impl/PojoConstructorProjectionDefinition.java

```diff
@@ -43,12 +44,14 @@
 		implements CompositeProjectionDefinition<T>, ToStringTreeAppendable {
 	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
 
+	private final PojoConstructorModel<T> constructor;
```

Comment:
You couldn't possibly know it, but this is actually a big no-no, as `PojoConstructorModel` is intended to be a boot-time-only thing, and `PojoConstructorProjectionDefinition` is used at runtime.

Why, you ask? `PojoConstructorModel` may have references to objects (e.g. `AbstractPojoHCAnnBootstrapIntrospector`) that we want to be garbage-collected after boot, because:

1. They may be large (a map of all types encountered so far and all their metadata, ...)
2. They expose operations that may not work fine in a native image, so we need to make sure they are not present in the heap after static init in Quarkus, so that GraalVM will infer "that class cannot work properly, but it's okay because I don't see any instance nor any way to create a new instance when executing the native image".

I suppose we could rely on `valueCreateHandle` to print *something*, but then the message would not be very user-friendly...

Maybe you can pre-build some representation of that constructor that can be used at runtime, similarly to what we do with `PojoModelPath`/`PojoModelPathPropertyNode`. Of course I'm not asking you to build a path of all constructors used in a projection definition (though that could be a nice follow-up issue), but just having a single object representing a single constructor and its parameters at runtime could be nice.

Regarding the constraints: you should preferably use `PojoRawTypeIdentifier` to represent types (use `PojoRawTypeModel.typeIdentifier()` to get an identifier), and avoid any references (direct or indirect) to `PojoTypeModel`, `PojoRawTypeModel`, `PojoConstructorModel` et al.